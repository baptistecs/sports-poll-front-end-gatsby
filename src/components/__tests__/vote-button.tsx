import React from "react"
import renderer from "react-test-renderer"
import { cleanup, render } from "@testing-library/react"
import VoteButton from "../vote-button"
import VOTE from "../../constants/vote"

describe("VoteButton", () => {
  it("renders correctly with label", () => {
    const component = renderer.create(
      <VoteButton
        id={1234}
        value={VOTE.DRAW}
        voteHandler={jest.fn()}
        label="Draw"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly without label", () => {
    const component = renderer.create(
      <VoteButton id={1234} value={VOTE.DRAW} voteHandler={jest.fn()} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup)

  it("call voteHandler on click", () => {
    const voteHandler = jest.fn()
    const { getByText } = render(
      <VoteButton
        id={1234}
        label="my button"
        value={VOTE.DRAW}
        voteHandler={voteHandler}
      />
    )
    const button = getByText(/my button/i)
    button.click()
    expect(voteHandler).toBeCalled()
  })
})
