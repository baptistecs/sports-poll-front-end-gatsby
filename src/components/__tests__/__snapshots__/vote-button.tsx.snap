// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteButton renders correctly with label 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "#fff",
      "border": "1px solid #fff",
      "borderRadius": 10,
      "color": "#000",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "100%",
      "overflowY": "auto",
      "width": "100%",
    }
  }
>
  Draw
</button>
`;

exports[`VoteButton renders correctly without label 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "#fff",
      "border": "1px solid #fff",
      "borderRadius": 10,
      "color": "#000",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "100%",
      "overflowY": "auto",
      "width": "100%",
    }
  }
>
  D
</button>
`;
